tests/integration_tests/functional/test_snapshot_not_losing_dirty_pages.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Test that the no dirty pages lost in case of error during snapshot creation."""
+
+import pytest
+import subprocess
+from pathlib import Path
+
+
+@pytest.fixture
+def mount_tmpfs_small(worker_id):
+    """Mount a small tmpfs and return its path"""
+    mnt_path = Path(f"/mnt/{worker_id}")
+    mnt_path.mkdir(parents=True)
+    subprocess.check_call(["mount", "-o", "size=512M", "-t", "tmpfs", "none", str(mnt_path)])
+    try:
+        yield mnt_path
+    finally:
+        subprocess.check_call(["umount", mnt_path])
+        mnt_path.rmdir()
+
+
+def test_diff_snapshot_works_after_error(
+    microvm_factory, guest_kernel_linux_5_10, rootfs_ubuntu_22, mount_tmpfs_small
+):
+    """
+    """
+
+    uvm = microvm_factory.build(
+        guest_kernel_linux_5_10,
+        rootfs_ubuntu_22,
+        jailer_kwargs={"chroot_base": mount_tmpfs_small},
+    )
+
+    vm_mem_size = 128
+    uvm.spawn()
+    uvm.basic_config(mem_size_mib=vm_mem_size, track_dirty_pages=True)
+    uvm.add_net_iface()
+    uvm.start()
+    uvm.ssh.run("true")
```

Comment:
We can use Taka's recent change `uvm.wait_for_up()`

Reply:
Thanks! I've updated to that